src/platforms/Gather/GatherConnectionForm.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  FormControl,
+  FormErrorMessage,
+  HStack,
+  Icon,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Link,
+  Stack,
+  Text,
+  Tooltip,
+} from "@chakra-ui/react"
+import { AddGatherFormType } from "components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/AddGatherPanel"
+import { Eye, EyeClosed, Question } from "phosphor-react"
+import { useState } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
+import GatherConnectionStatusAlert from "./GatherConnectionStatusAlert"
+import useGatherAccess from "./hooks/useGatherAccess"
+
+const GatherConnectionForm = () => {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext<AddGatherFormType>()
+
+  const [showApiKey, setShowApiKey] = useState(false)
+
+  const [gatherApiKey, gatherSpaceId] = useWatch({
+    name: ["gatherApiKey", "gatherSpaceId"],
+  })
+  const {
+    success: accessSuccess,
+    isLoading: accessLoading,
+    error: accessError,
+  } = useGatherAccess(
+    !errors?.gatherApiKey && gatherApiKey,
+    !errors?.gatherSpaceId && gatherSpaceId
+  )
+
+  return (
+    <>
+      <GatherConnectionStatusAlert
+        isLoading={accessLoading}
+        success={accessSuccess}
+        error={accessError}
+      />
+
+      <FormControl pt={{ md: 0.5 }} isInvalid={!!errors?.gatherApiKey}>
+        <Stack>
+          <HStack>
+            <Text as="span">API key:</Text>
+            <Tooltip
+              label="Used to securely communicate with Gather"
+              placement="top"
+              hasArrow
+            >
+              <Icon as={Question} color="GrayText" />
+            </Tooltip>
+          </HStack>
+          <InputGroup>
+            <Input
+              autoComplete="off"
+              type={showApiKey ? "text" : "password"}
+              {...register("gatherApiKey", {
+                required: "This field is required",
+              })}
+            />
+            <InputRightElement onClick={() => setShowApiKey(!showApiKey)}>
+              {showApiKey ? <EyeClosed /> : <Eye />}
```

Comment:
Would be great to wrap the icons in an IconButton so we have hover/focus states and they're focusable. Or at least set a cursor: pointer style